protocols/dcutr/src/behaviour_impl.rs

```diff
@@ -148,40 +157,57 @@ impl Behaviour {
     fn on_dial_failure(
         &mut self,
         DialFailure {
-            peer_id, handler, ..
-        }: DialFailure<<Self as NetworkBehaviour>::ConnectionHandler>,
+            peer_id,
+            connection_id: failed_direct_connection,
+            ..
+        }: DialFailure,
     ) {
-        if let handler::Prototype::DirectConnection {
-            relayed_connection_id,
-            role: handler::Role::Initiator { attempt },
-        } = handler
+        let peer_id = if let Some(peer_id) = peer_id {
+            peer_id
+        } else {
+            return;
+        };
```

Comment:
nit: I think we can do:
```suggestion
        let Some(peer_id) = peer_id {
            peer_id
        } else {
            return;
        };
```
if we bump `rust-version` on `Cargo.toml` to [`1.65`](https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#let-else-statements) :stuck_out_tongue:

Reply:
I didn't want to bump MSRV just for that but I am certainly looking forward to `let-else` :)